prdoc/pr_3685.prdoc

```diff
@@ -0,0 +1,300 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: FRAME Reintroduce `TransactionExtension` as a replacement for `SignedExtension`
```

Comment:
None of this seems to be in the changelog of `stable2412` :exploding_head:

This is such a massive change with numerous breaking API changes, it must be included in the release notes.

Reply:
The issue in the changelog should def be addressed, already reported internally. 

But if you don't want to upgrade to `TransactionExtension`, how is this a breaking change?

Reply:
I just commented on this line, I didn't mean the `TransactionExtension` specifically.

This PR broke a large number of APIs from renaming of data structures to straight removal. Upgrade to `polkadot-stable2412-1` will require major implementation changes for us.

For example unsigned extrinsics is something that our custom consensus implementation relies on right now, but despite documented as "deprecated" here, the API to submit them seems to be gone in the stable release, preventing us from upgrading in stages and handling deprecations in a reasonable amount of time. Or maybe there API is there, but is in a different place and/or called differently, I have no idea right now without any upgrade documentation on this.

Reply:
@nazar-pc it is still there, but the naming is totally wrong: https://github.com/paritytech/polkadot-sdk/pull/3685/files#r1721106205

I complained, but apparently it did not get fixed...

Reply:
https://github.com/paritytech/polkadot-sdk/blob/b600af050d6b6c8da59ae2a2a793ee2d8827ab1e/substrate/frame/system/src/offchain.rs#L492 direct link to the file.

Reply:
Inherent is a different thing, at least it was before `stable2412`. We use inherents too, but here I do mean unsigned extrinsic that is being gossiped across the network.

Reply:
As I said, the new naming is not that great. This is the trait you are searching for.

Reply:
yes conversation was here: https://github.com/paritytech/polkadot-sdk/pull/3685/files#r1721106205

maybe we can at least improve the doc: https://github.com/paritytech/polkadot-sdk/pull/7597

Reply:
BTW @nazar-pc the changelog was updated! This should not have been missed (there was a bug)

Reply:
Thank you!